src/openforms/data_removal/tests/test_tasks.py

```diff
@@ -58,6 +58,34 @@ def test_successful_submissions_correctly_deleted(self):
         with self.assertRaises(ObjectDoesNotExist):
             submission_to_be_deleted.refresh_from_db()
 
+    def test_successful_submissions_correctly_deleted_the_same_day_when_form_removal_limit_is_0(
+        self,
+    ):
+        form = FormFactory.create(
+            successful_submissions_removal_limit=0,
+            incomplete_submissions_removal_limit=7,
+            errored_submissions_removal_limit=7,
+            all_submissions_removal_limit=7,
+        )
+
+        with freeze_time(lambda: datetime(2024, 11, 12, 18)):
```

Comment:
what's up with the lambda? Is there a reason you can't do:

```py
with freeze_time("2024-11-12T18:00:00+01:00"):
```

furthermore, *never ever* use naive datetimes in python/django (yours is naive, because no `tzinfo` has been specified). Generally you want to use the django wrappers defined in `django.utils.timezone`.

Reply:
Ah yeah, that lambda isn't needed 👍 

Ah okay, check. I'll keep it in mind